core/src/oracle_state.rs

```diff
@@ -635,3 +588,68 @@ pub fn register_and_save_scans() -> std::result::Result<(), Error> {
     }
     Ok(())
 }
+
+/// Registers and saves scans to `scanIDs.json` as well as performing wallet rescanning.
+///
+/// WARNING: will overwrite existing `scanIDs.json`!
+fn register_and_save_scans_inner(
+    oracle_pool_participant_token_id: &TokenId,
```

Comment:
Do we need these arguments at all? I mean, they are all loaded from oracle config at the call site and are not used anywhere else. I think they could all be defined inside this function.

Reply:
Oh yes of course, sorry about that. It's done.